File: KGH/Greedy/prog_level1_체육복.py
def solution(n, lost, reserve):
    answer = 0
    lost.sort()
    reserve.sort()

    new_lost = []
    for x in lost:
        if x in reserve:    # lost의 원소 x가 reserve에 있다면
            reserve.remove(x)
        else:
            new_lost.append(x)
    lost = new_lost

    
    for i in reserve :
        if (i-1) in lost : 
            lost.remove(i-1)

        elif (i+1) in lost :
            lost.remove(i+1)

    answer = n - len(lost)

File: PJH/그리디/prog_level1_체육복.swift
//
//  prog_level1_체육복.swift
//  PJH
//
//  Created by ParkJunHyuk on 2023/11/27.
//

import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var lostSet = Set(lost).subtracting(reserve)
    let reserveSet = Set(reserve).subtracting(lost)
    
    for reserve in reserveSet {
        if lostSet.contains(reserve - 1) {
            lostSet.remove(reserve - 1)
            continue
        }
        if lostSet.contains(reserve + 1) {
            lostSet.remove(reserve + 1)
        }
    }
    return (n - lostSet.count)
}
